server/src/main/java/io/crate/metadata/doc/DocTableInfo.java

```diff
@@ -733,6 +729,32 @@ private void updateParentsInnerTypes(ColumnIdent column,
         }
     }
 
+    public DocTableInfo dropConstraint(String constraint) {
+        List<CheckConstraint<Symbol>> newConstraints = checkConstraints.stream()
+            .filter(x -> !x.name().equals(constraint))
+            .toList();
+        if (newConstraints.size() == checkConstraints.size()) {
+            return this;
+        }
```

Comment:
We already analyze the constraint so I think it should always be false.

Reply:
There could still be a race. The check here is cheap and can avoid a cluster state update which would be expensive compared to the check here